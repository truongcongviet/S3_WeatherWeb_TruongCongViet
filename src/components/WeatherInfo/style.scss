.weatherInfo {
    color: var(--white);
    height: 60vh;
    background-image: linear-gradient(180deg, var(--blue-sky), var(--blue-light), var(--blue-stable));
}
