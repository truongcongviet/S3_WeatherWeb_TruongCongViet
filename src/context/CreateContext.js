import { createContext, useCallback, useContext, useReducer, Fragment } from 'react';
import { weatherReducer, initialState } from './reducers/weatherReducer'
const CreateContext = createContext({});

export const useAppContext = () => {
    const { data, setData } = useContext(CreateContext)
    const dispatch = useCallback(
        ({ type, data }) => {
            setData({ data, type })
        },
        [setData]
    )
    return [data, dispatch]
}

export const WeatherProvider = props => {
    const { children } = props;
    const [data, dispatch] = useReducer(weatherReducer, initialState);
    return (
        <CreateContext.Provider value={{ data, setData: dispatch }}>
            <Fragment>
                {children}
            </Fragment>

        </CreateContext.Provider>
    )
}

export default CreateContext;