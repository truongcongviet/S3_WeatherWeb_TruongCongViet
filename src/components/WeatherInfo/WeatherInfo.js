import WeatherHeader from "../WeatherHeader/WeatherHeader";
import WeatherDetail from "../WeatherDetail/WeatherDetail";
import "./style.scss";
import weatherServices from "../../services/weather.services";
import { GET_DATA_WEATHER } from "../../context/constant/actionGetDataWeather";
import { useAppContext } from "../../context/CreateContext";
import { useEffect, useState } from "react";
const WeatherInfo = () => {
  const dispatch = useAppContext()[1]
  useEffect(() => {
    getCurrentLocation()
  }, [])

  //get curent location
  const getCurrentLocation = () => {
    navigator.geolocation.getCurrentPosition((position) => {
      getData({
        lat: position.coords.latitude,
        lon: position.coords.longitude, 
      })
    }, error => {
      alert('Cannot get current location')
      console.log('get location error:', error)
    });
  }

  //get Weather API from location
  const getData = (location) => {
    weatherServices.oneCall({ lat: location.lat, lon: location.lon }).then(res => {
      dispatch({ type: GET_DATA_WEATHER, data: res.data });
    });
  }

  return (
    <div className="weatherInfo" role="weather-test">
      <WeatherHeader></WeatherHeader>
      <WeatherDetail></WeatherDetail>
    </div>
  )
}

export default WeatherInfo;