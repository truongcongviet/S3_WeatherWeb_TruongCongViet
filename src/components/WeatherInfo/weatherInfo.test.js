import React from 'react';
import WeatherInfo from './WeatherInfo';
import { render } from '@testing-library/react';
import { WeatherProvider } from '../../context/CreateContext';

const wrapper = render(<WeatherProvider><WeatherInfo /></WeatherProvider>);
describe('WeatherInfo Component test', () => {
    test("expect to render WeatherInfo component", () => {
        const role = wrapper.getByRole('weather-test');
        expect(role).toHaveClass('weatherInfo');
        expect(wrapper).toMatchSnapshot();
    })
})