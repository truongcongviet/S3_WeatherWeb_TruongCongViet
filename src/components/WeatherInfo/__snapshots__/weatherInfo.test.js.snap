// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherInfo Component test expect to render WeatherInfo component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="weatherInfo"
        role="weather-test"
      >
        <div
          class="weatherHeader"
          role="header"
        >
          <div
            class="menu-icon"
          >
            <i
              class="bi bi-list h1"
            />
          </div>
          <div
            class="get-location"
          >
            <div>
              myENV
            </div>
            <select
              class="custom-select-cities"
            >
              <option
                class="city"
                value="0"
              >
                Hồ Chí Minh
              </option>
              <option
                class="city"
                value="1"
              >
                Singapore
              </option>
              <option
                class="city"
                value="2"
              >
                Jerusalem
              </option>
              <option
                class="city"
                value="3"
              >
                California
              </option>
              <option
                class="city"
                value="4"
              >
                AlasKa
              </option>
            </select>
          </div>
          <div
            class="bell-icon"
          >
            <i
              class="bi bi-bell-fill h2"
            />
          </div>
        </div>
        <div
          class="weatherDetail"
        >
          <div
            class="cloudyInfo"
          >
            <div
              class="cloud-icon p-2 mb-4"
            >
              <i
                class="bi bi-cloud-minus"
              />
            </div>
            <div
              class="weatherData p-2"
            >
              <div
                class="h2"
              />
              <div
                class="d-flex h4 justify-content-between"
              >
                <div
                  class=""
                >
                  <i
                    class="bi bi-thermometer-half"
                  />
                   
                   °C
                </div>
                <div>
                  <i
                    class="bi bi-droplet"
                  />
                   
                   %
                </div>
              </div>
            </div>
          </div>
          <ul
            class="rainingInfo p-0"
          >
            <li>
              <h5>
                PSI
              </h5>
              <div
                class="feel_like"
              />
              <p>
                good
              </p>
            </li>
            <li>
              <h5>
                RAIN
              </h5>
              <div>
                0
              </div>
              <p>
                mm
              </p>
            </li>
            <li>
              <h5>
                DENGUE
              </h5>
              <div
                class="dengue"
              />
            </li>
            <li>
              <i
                class="bi bi-plus-circle h1 text-dark"
              />
              <p
                class="text-dark font-weight-bold h4"
              >
                add
              </p>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="weatherInfo"
      role="weather-test"
    >
      <div
        class="weatherHeader"
        role="header"
      >
        <div
          class="menu-icon"
        >
          <i
            class="bi bi-list h1"
          />
        </div>
        <div
          class="get-location"
        >
          <div>
            myENV
          </div>
          <select
            class="custom-select-cities"
          >
            <option
              class="city"
              value="0"
            >
              Hồ Chí Minh
            </option>
            <option
              class="city"
              value="1"
            >
              Singapore
            </option>
            <option
              class="city"
              value="2"
            >
              Jerusalem
            </option>
            <option
              class="city"
              value="3"
            >
              California
            </option>
            <option
              class="city"
              value="4"
            >
              AlasKa
            </option>
          </select>
        </div>
        <div
          class="bell-icon"
        >
          <i
            class="bi bi-bell-fill h2"
          />
        </div>
      </div>
      <div
        class="weatherDetail"
      >
        <div
          class="cloudyInfo"
        >
          <div
            class="cloud-icon p-2 mb-4"
          >
            <i
              class="bi bi-cloud-minus"
            />
          </div>
          <div
            class="weatherData p-2"
          >
            <div
              class="h2"
            />
            <div
              class="d-flex h4 justify-content-between"
            >
              <div
                class=""
              >
                <i
                  class="bi bi-thermometer-half"
                />
                 
                 °C
              </div>
              <div>
                <i
                  class="bi bi-droplet"
                />
                 
                 %
              </div>
            </div>
          </div>
        </div>
        <ul
          class="rainingInfo p-0"
        >
          <li>
            <h5>
              PSI
            </h5>
            <div
              class="feel_like"
            />
            <p>
              good
            </p>
          </li>
          <li>
            <h5>
              RAIN
            </h5>
            <div>
              0
            </div>
            <p>
              mm
            </p>
          </li>
          <li>
            <h5>
              DENGUE
            </h5>
            <div
              class="dengue"
            />
          </li>
          <li>
            <i
              class="bi bi-plus-circle h1 text-dark"
            />
            <p
              class="text-dark font-weight-bold h4"
            >
              add
            </p>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
